### PR TITLE
Add NetworkPolicy for Prometheus

### PR DESCRIPTION
**Pull request by ozdanborne**

### CODE REVIEW

File: stable/prometheus/templates/alertmanager-networkpolicy.yaml [hunk 1]
```diff
@@ -0,0 +1,20 @@
+{{- if .Values.networkPolicy.enabled }}
+apiVersion: {{ .Values.networkPolicy.apiVersion }}
```
Comment (mgoodness) [inline stable/prometheus/templates/alertmanager-networkpolicy.yaml]: Let's take this off users' shoulders and use Helm's `.Capabilities` object to set the API version accordingly.

File: stable/prometheus/templates/alertmanager-networkpolicy.yaml [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+kind: NetworkPolicy
+metadata:
+  name: {{ template "prometheus.alertmanager.fullname" . }}
```
Comment (mgoodness) [inline stable/prometheus/templates/alertmanager-networkpolicy.yaml]: Add standard labels (`app`, `component`, `chart`, `release`, `heritage`).

File: stable/prometheus/templates/alertmanager-networkpolicy.yaml [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+    matchLabels:
+      app: {{ template "prometheus.name" . }}
+      component: "{{ .Values.alertmanager.name }}"
+      release: {{ .Release.Name }}
+  ingress:
+    - from:
+      - podSelector:
+          matchLabels:
+            release: {{ .Release.Name }}
+            component: {{ .Values.server.name }}
```
Comment (mgoodness) [inline stable/prometheus/templates/alertmanager-networkpolicy.yaml]: Needs quotes around the value.

File: stable/prometheus/Chart.yaml [hunk 4]
```diff
@@ -1,5 +1,5 @@
 name: prometheus
-version: 3.0.2
+version: 3.0.3
```
Comment (mgoodness) [inline stable/prometheus/Chart.yaml]: We're adding functionality, so let's make this a minor (not patch) version bump.

File: stable/prometheus/templates/kube-state-metrics-networkpolicy.yaml [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+      release: {{ .Release.Name }}
+  ingress:
+  - from:
+    - podSelector:
+        matchLabels:
+          release: {{ .Release.Name }}
+          component: {{ .Values.server.name }}
+  - ports:
+    - port: 8080
+{{- end }}
```
Comment (mgoodness) [inline stable/prometheus/templates/kube-state-metrics-networkpolicy.yaml]: Comments from `alertmanager-networkpolicy.yaml` apply here.

File: stable/prometheus/templates/server-networkpolicy.yaml [hunk 6]
```diff
@@ -0,0 +1,15 @@
...
+spec:
+  podSelector:
+    matchLabels:
+      app: {{ template "prometheus.name" . }}
+      component: "{{ .Values.server.name }}"
+      release: {{ .Release.Name }}
+  ingress:
+    - ports:
+      - port: 9090
+{{- end }}
```
Comment (mgoodness) [inline stable/prometheus/templates/server-networkpolicy.yaml:21]: Comments from `alertmanager-networkpolicy.yaml` apply here as well.

**Review by mgoodness — CHANGES_REQUESTED** [review]
Looks pretty good, but I think we can eliminate the need to specify `apiVersion`. Don't forget to rebase!

**Comment by mgoodness** [comment]
@k8s-bot ok to test

**Comment by ozdanborne** [comment]
@mgoodness I've been trying to implement `.Capabilities` to automatically provide the correct networkPolicy apiVersion, but have been running into a lot of issues.

For starters, comparator functions (`ge`, `le`) do not accept strings. Only `int`. Which is quite confusing as [there's](https://github.com/kubernetes/charts/blob/3e43eb735c2da44565aea81bd65bb155fbaa2272/stable/traefik/templates/deployment.yaml#L17) [a](https://github.com/kubernetes/charts/blob/74dda6cdc563d27935daccb2d97336bc5625ae66/stable/kube2iam/templates/daemonset.yaml#L56) [few](https://github.com/kubernetes/charts/blob/f2e6b169cbfb1c5f131042f14fec9f0f8794bb8e/stable/spotify-docker-gc/templates/daemonset.yaml#L6) places where charts are comparing strings.

I can get past that syntax error by casting them to integers. But then when I try to deploy my chart, `.Capabilities.KubeVersion.Minor` and `.Major` are always nil. 

Any pointers on what's going on? I've tried using helm v2.4.2 and v2.5.1

Edit: Forgot to say thank you for taking a look and providing those suggestions :smile:

**Comment by ozdanborne** [comment]
Thanks for helping me to realize I wasn't passing the current context to my template functions @mgoodness .

I think I've addressed all feedback at this point. Look good? feedback is in 2nd commit. I can squash when ready to merge

**Review by mgoodness — APPROVED** [review]

**Pull request merged**